client/js/widgets/basicwidget.js

```diff
@@ -63,11 +63,16 @@ class BasicWidget extends Widget {
     return p;
   }
 
-  flip(setFlip) {
+  flip(setFlip, faceCycle) {
     if(setFlip !== undefined && setFlip !== null)
       this.p('activeFace', setFlip);
-    else
-      this.p('activeFace', Math.floor(this.p('activeFace') + (this.p('faceCycle') == 'random' ? Math.random()*99999 : 1)) % this.p('faces').length);
+    else {
+      const fC = (faceCycle == 'previous' || faceCycle == 'random') ? faceCycle : this.p('faceCycle');
```

Comment:
- [ ] For `faceCycle == 'ordered'` this ignores the provided `faceCycle` and falls back to `this.p('faceCycle')` which could be something else.